components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1326 @@
+
+  wsubmin_ = m_params.get<double>("wsubmin");
+  top_lev_ = m_params.get<int>("top_level_mam4xx");
+
+  // ------------------------------------------------------------------------
+  // Input fields read in from IC file, namelist or other processes
+  // ------------------------------------------------------------------------
+  w_sec_mid_    = get_field_in("w_variance").get_view<const Real **>();
+  dgnum_        = get_field_in("dgnum").get_view<const Real ***>();
+  liqcldf_      = get_field_in("cldfrac_liq").get_view<const Real **>();
+  liqcldf_prev_ = get_field_in("cldfrac_liq_prev").get_view<const Real **>();
+  kvh_mid_      = get_field_in("eddy_diff_heat").get_view<const Real **>();
+
+  // store fields only to be converted to dry mmrs in wet_atm_
+  wet_atm_.qv = get_field_in("qv").get_view<const Real **>();
+  wet_atm_.qc = get_field_in("qc").get_view<const Real **>();
+  wet_atm_.nc = get_field_in("nc").get_view<const Real **>();
+  wet_atm_.qi = get_field_in("qi").get_view<const Real **>();
+  wet_atm_.ni = get_field_in("ni").get_view<const Real **>();
+
+  // store rest fo the atm fields in dry_atm_in
+  dry_atm_.T_mid = get_field_in("T_mid").get_view<const Real **>();
+  dry_atm_.p_mid = get_field_in("p_mid").get_view<const Real **>();
+  dry_atm_.p_int = get_field_in("p_int").get_view<const Real **>();
+  dry_atm_.p_del = get_field_in("pseudo_density").get_view<const Real **>();
+  dry_atm_.omega = get_field_in("omega").get_view<const Real **>();
+
+  // store fields converted to dry mmr from wet mmr in dry_atm_
+  dry_atm_.qv = buffer_.qv_dry;
+  dry_atm_.qc = buffer_.qc_dry;
+  dry_atm_.nc = buffer_.nc_dry;
+  dry_atm_.qi = buffer_.qi_dry;
+  dry_atm_.ni = buffer_.ni_dry;
+
+  // pbl_height
+  dry_atm_.pblh = get_field_in("pbl_height").get_view<const Real *>();
+
+  // geometric thickness of layers (m)
+  dry_atm_.dz = buffer_.dz;
+
+  // geopotential height above surface at interface levels (m)
+  dry_atm_.z_iface = buffer_.z_iface;
+
+  // geopotential height above surface at mid levels (m)
+  dry_atm_.z_mid = buffer_.z_mid;
+
+  // total cloud fraction
+  dry_atm_.cldfrac = get_field_in("cldfrac_tot").get_view<const Real **>();
+
+  // computed updraft velocity
+  dry_atm_.w_updraft = buffer_.w_updraft;
+
+  // ------------------------------------------------------------------------
+  // Output fields to be used by other processes
+  // ------------------------------------------------------------------------
+  // ice nucleation output
+  naai_ = get_field_out("ni_activated").get_view<Real **>();
+
+  // droplet activation output
+  tendnd_ = get_field_out("nc_nuceat_tend").get_view<Real **>();
+
+  // interstitial and cloudborne aerosol tracers of interest: mass (q) and
+  // number (n) mixing ratios
+  for(int m = 0; m < mam_coupling::num_aero_modes(); ++m) {
+    // interstitial aerosol tracers of interest: number (n) mixing ratios
+    const char *int_nmr_field_name = mam_coupling::int_aero_nmr_field_name(m);
```

Comment:
Any reason to use C strings instead of std::string?

Reply:
@bartgol, will std::string produce the warning of calling host function from device?

Reply:
This is not device code though, so std::string is fine.

Reply:
I will fix it in a follow-up PR (see issue #2850 )